SynCommons.pas

```diff
@@ -24209,12 +24209,21 @@ function ToUTF8(Value: Int64): RawUTF8;
 {$endif CPU64}
 
 function ToUTF8(Value: PtrInt): RawUTF8;
+{$ifdef CPU64}
+var tmp: array[0..23] of AnsiChar;
+    P: PAnsiChar;
+begin
+  P := StrInt64(@tmp[23],Value);
+  FastSetString(result,P,@tmp[23]-P);
+end;
+{$else}
 var tmp: array[0..15] of AnsiChar;

```

Comment:
replacing it with `var tmp: array[0..23] of AnsiChar;` sounds simplier thand a ifdef/endif since StrInt64=StrInt32

Reply:
Let's keep StrInt64 where
 - call to StrInt32 inside Int64 convertion is confused
 - StrInt32 is implemented in assembler and it's very unclear what it can convert both Int32 and Int64